raiden-ts/tests/unit/epics/transport.spec.ts

```diff
@@ -1828,3 +1836,1716 @@ describe('transport epic', () => {
     });
   });
 });
+
+describe('transport epic', () => {
+  const matrixServer = 'matrix.raiden.test'; // define later in fixture constant
+  const partnerRoomId = `!partnerRoomId:${matrixServer}`;
+  const processed: Processed = {
+    type: MessageType.PROCESSED,
+    message_identifier: makeMessageId(),
+  };
+  const fetch = jest.fn(async () => ({
+    ok: true,
+    status: 200,
+    text: jest.fn(async () => `- ${matrixServer}`),
+  }));
+  Object.assign(global, { fetch });
+
+  describe('initMatrixEpic', () => {
+    test('matrix stored setup', async () => {
+      expect.assertions(2);
+      const [raiden] = await makeRaidens(1);
+      const matrix = (await raiden.deps.matrix$.toPromise()) as jest.Mocked<MatrixClient>;
+
+      const userId = matrix.getUserId()!;
+      expect(raiden.output).toContainEqual(
+        matrixSetup({
+          server: `https://${matrix.getHomeserverUrl()}`,
+          setup: {
+            userId,
+            accessToken: expect.any(String),
+            deviceId: expect.any(String),
+            displayName: expect.any(String),
+          },
+        }),
+      );
+      // ensure if stored setup works, servers list don't need to be fetched
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    test('matrix server config set without stored setup', async () => {
+      expect.assertions(2);
+      const [raiden] = await makeRaidens(1);
+      raiden.store.dispatch(raidenConfigUpdate({ matrixServer }));
+      const matrix = (await raiden.deps.matrix$.toPromise()) as jest.Mocked<MatrixClient>;
+      const userId = matrix.getUserId()!;
+
+      expect(raiden.output).toContainEqual(
+        matrixSetup({
+          server: `https://${matrix.getHomeserverUrl()}`,
+          setup: {
+            userId,
+            accessToken: expect.any(String),
+            deviceId: expect.any(String),
+            displayName: expect.any(String),
+          },
+        }),
+      );
+      expect(fetch).not.toHaveBeenCalled();
+    });
+
+    test('matrix fetch servers list', async () => {
+      expect.assertions(2);
+
+      jest.setTimeout(30000);
```

Comment:
Don't set timeout inside a test. If you need to set it there, you must pass it as 3rd param to `test` declaration. But 30s is too much for a test anyway, you probably should decrease `httpTimeout`/`pollingInterval` config params if you have some timeout taking longer than it should inside your test, instead of increasing test timeout.

Reply:
Just made required changes